ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureCloudDestination.java

```diff
@@ -457,6 +458,28 @@ AzureBlobDataAccessor getAzureBlobDataAccessor() {
     return azureBlobDataAccessor;
   }
 
+  /**
+   * Construct a {@link CloudStorageException} from a root cause exception.
+   * @param message the exception message.
+   * @param e the root cause exception.
+   * @return the {@link CloudStorageException}.
+   */
+  private static final CloudStorageException toCloudStorageException(String message, Exception e) {
+    boolean isRetryable = false;
+    Long retryDelayMs = null;
+    int statusCode = -1;
+    if (e instanceof BlobStorageException) {
+      statusCode = ((BlobStorageException) e).getStatusCode();
+    }
+    else if (e instanceof DocumentClientException) {
+      statusCode = ((DocumentClientException) e).getStatusCode();
+      retryDelayMs = ((DocumentClientException) e).getRetryAfterInMilliseconds();
+    }
+    // Everything is retryable except NOT_FOUND
+    isRetryable = (statusCode != HttpConstants.StatusCodes.NOTFOUND);
```

Comment:
we can define isRetryable here instead of above.

Reply:
Good point.  Is it okay to fix this in a follow up PR?  This class has work in progress for another ticket.